Mage.Sets/src/mage/cards/f/FireGiantsFury.java

```diff
@@ -0,0 +1,242 @@
+package mage.cards.f;
+
+import mage.abilities.Ability;
+import mage.abilities.DelayedTriggeredAbility;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BoostTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.keyword.TrampleAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ *
+ * @author weirddan455
+ */
+public final class FireGiantsFury extends CardImpl {
+
+    private static final FilterControlledCreaturePermanent filter = new FilterControlledCreaturePermanent(SubType.GIANT);
+
+    public FireGiantsFury(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{1}{R}");
+
+        // Target Giant you control gets +2/+2 and gains trample until end of turn.
+        Effect effect = new BoostTargetEffect(2, 2, Duration.EndOfTurn);
+        effect.setText("Target Giant you control gets +2/+2");
+        this.getSpellAbility().addEffect(effect);
+
+        effect = new GainAbilityTargetEffect(TrampleAbility.getInstance(), Duration.EndOfTurn);
+        effect.setText("and gains trample until end of turn");
+        this.getSpellAbility().addEffect(effect);
+
+        // Whenever it deals combat damage to a player this turn, exile that many cards from the top of your library. Until the end of your next turn, you may play those cards.
+        effect = new FireGiantsFuryEffect();
+        this.getSpellAbility().addEffect(effect);
+
+        this.getSpellAbility().addTarget(new TargetPermanent(filter));
+    }
+
+    private FireGiantsFury(final FireGiantsFury card) {
+        super(card);
+    }
+
+    @Override
+    public FireGiantsFury copy() {
+        return new FireGiantsFury(this);
+    }
+}
+
+class FireGiantsFuryEffect extends OneShotEffect {
+
+    public FireGiantsFuryEffect() {
+        super(Outcome.PlayForFree);
+        this.staticText = "Whenever it deals combat damage to a player this turn, exile that many cards from the top of your library. Until the end of your next turn, you may play those cards";
```

Comment:
You should add `<br>` to the beginning of this to add a line break.

Reply:
It's a one line ability in printed card. So no needs in line break.

Reply:
Oh ok then scratch that

Reply:
Thanks.  I got it to check for combat damage correctly now.  However, I can't seem to pass the damage in `checkTrigger` like you said.  I made a damage variable:

```
class FireGiantsFuryDelayedEffect extends OneShotEffect {

    private int damage = 0
```

Created a setDamage method:

```
public void setDamage(int damage) {
    this.damage = damage;
}
```

Then call it from `checkTrigger`:

```
public boolean checkTrigger(GameEvent event, Game game) {
    if (event.getSourceId().equals(target) && ((DamagedEvent) event).isCombatDamage()) {
        FireGiantsFuryDelayedEffect effect = (FireGiantsFuryDelayedEffect) this.getEffects().get(0);
        effect.setDamage(event.getAmount());
        return true;
```

I added some debugging code and `setDamage` is being called and is being passed the correct number but for some reason when `apply` runs and checks `damage`, the value is 0.

I feel like I'm missing something simple here but having trouble figuring out what.

Reply:
take a look at some of the other combat damage triggered abilities

Reply:
Do you know of any example cards that use a delayed trigger on combat damage?  I looked at the code for Garruk's Harbinger and it just does

`Integer damage = (Integer) getValue("damage");`

inside `apply` but this doesn't work for a delayed trigger.

I'm coming up empty searching ScryFall for cards with similar wording.  The card seems somewhat unique in that way.

The watcher I made does work though.  Is there a reason why that's not the correct way to handle this?

Reply:
The problem with the watcher is it doesn't work if some of the damage is redirected to a different target. Take a look at the `checkTrigger` method for `DealsCombatDamageToAPlayerTriggeredAbility`.

Reply:
[Hunter’s Insight](https://scryfall.com/card/cmr/232/hunters-insight)

Choose target creature you control. Whenever that creature deals combat damage to a player or planeswalker this turn, draw that many cards.

Reply:
yeah, that way works too

Reply:
Oh, thanks that was helpful.  I think I've got it working now.  Let me just clean up the code and I'll push another commit with the changes.